databaseManager.class.php

```diff
@@ -0,0 +1,157 @@
+
+        /**
+         * Connect database
+         */
+        abstract protected function openConnection();
+
+        /**
+         * Returns rows from the database based on the conditions
+         * @param string name of the table
+         * @param array select, where, order_by, limit and return_type conditions
+         */
+        abstract protected function selectData($table, $conditions = array());
+
+        /**
+         * Insert data into the database
+         * @param string name of the table
+         * @param array the data for inserting into the table
+         */
+        abstract protected function insertData($table,$data);
+        
+        /**
+         * Update data into the database
+         * @param string name of the table
+         * @param array the data for updating into the table
+         * @param array where condition on updating data
+         */
+        abstract protected function updateData($table, $data, $conditions);
+
+        /**
+         * Delete data from the database
+         * @param string name of the table
+         * @param array where condition on deleting data
+         */
+        abstract protected function deleteData($table, $conditions);
+
+        /**
+         * Constructor
+         */
+        protected function __construct($dbDriver, $host, $port, $username, $password, $dbName, $unixSocket, $charset)
+        {
+            $this->dbDriver = $dbDriver;
+            $this->host = $host;
+            $this->port = $port;
+            $this->username = $username;
+            $this->password = $password;
+            $this->dbName = $dbName;
+            $this->unixSocket = $unixSocket;
+            $this->charset = $charset;
+        }
+
+        /**
+         * @return bool
+         */
+        public function isConnected()
+        {
+            return ($this->connection instanceof PDO);
+        }
+
+        /**
+         * Kill connection
+         */
+        public function closeConnection()
+        {
+            $this->connection = null;
+        }
+
+        /**
+         * Start transaction
+         */
+        public function startTransaction()
+        {
+            $this->connection->beginTransaction();
+        }
+
+        /**
+         * Commit the changes
+         */
+        public function commitTransaction()
+        {
+            $this->connection->commit();
+        }
+
+        /**
+         * Rollback the changes
+         */
+        public function rollbackTransaction()
+        {
+            $this->connection->rollBack();
+        }
+
+        /**
+         * Store cache file in cache folder
+         * @param string name of cache file
+         * @param array data from select query
+         */
+        public function storeCacheFile($sqlCacheName, $data)
+        {
+            /* Write data to file */
+            $filew = fopen("cache/" . $sqlCacheName, 'w');
+            fwrite($filew, print_r($data, true));
+            fclose($filew);
```

Comment:
@lumakos Can you tell some common problems with file cache?

Reply:
First of all , i think the best solution for query caching is memcache. About file cache, the first common problem is the file's permissions, also i sould declare a variable which stores the location of the cache file, current time and the time when the cache file was last modified.

Reply:
@lumakos What, If the application runs on a multiple-web server? The cached data somehow should be shared among all these servers.

Reply:
@tavantzo I think that servers will be confused because cache files do not have an ID or sth to be recognized by servers.